workflow/controller/exec_control.go

```diff
@@ -81,7 +81,9 @@ func (woc *wfOperationCtx) handleExecutionControlError(nodeID string, wfNodesLoc
 	// then need to fail child nodes so they will not hang in Pending after pod deletion
 	for _, nodeID := range node.Children {
 		child := woc.wf.Status.Nodes[nodeID]
-		woc.markNodePhase(child.Name, wfv1.NodeFailed, errorMsg)
```

Comment:
This code is changing the Succeeded  to failed. 
controller | time="2022-03-05T06:09:55.576Z" level=info msg="node stop-terminate-f4dqc-1853455048 phase Succeeded -> Failed" namespace=argo workflow=stop-terminate-f4dqc